activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1195,6 +1251,37 @@ def build_select(arel)
         end
       end
 
+      def build_with(arel)
+        return if with_values.empty?
```

Comment:
Is this needed? If `with_values` is empty, `with_values.map` will result into empty array. You just need to check in the end for `with_statements.any?` to apply `arel.with` if present.

Reply:
I think guard clause works great here.

It would also work without it but you need to add `any?` check anyways and all the calls before will be called also.

Reply:
There is no guard clause used for another `build_*` methods, that is the idea to stick closer to current code.